File: app/src/main/java/com/wzf/wucarryme/common/utils/StringUtil.java
package com.wzf.wucarryme.common.utils;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import android.support.annotation.NonNull;

/**
 * @author wzf
 * @date 2018/3/22
 */

public class StringUtil {

    public static String banKuai(String str) {

        str = str.replaceAll("类", "");
        str = str.replaceAll("概念", "");
        str = str.replaceAll("房地产", "地产");
        str = str.replaceAll("次新", "新");
        int index = str.indexOf("股");
        if (index > -1) {
            str = str.substring(0, index);
        }
        return str;
    }

    public static String getBlogTime(String p) {
        String blogTime = "";
        int endIndex = p.contains(" ") ? p.indexOf(" ") : p.indexOf(160);
        blogTime = p.substring(0, endIndex);
        return blogTime;
    }

    public static List<String> commaStr2List(String commaStr) {
        return Arrays.asList(commaStr.split(","));
    }

    public static String list2commaStr(List<String> list) {
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < list.size(); i++) {
            result.append(list.get(i)).append(",");
        }
        if (result.toString().endsWith(",")) {
            result = new StringBuilder(result.substring(0, result.length() - 1));
        }
        return result.toString();
    }

    @NonNull
    public static String formatCNName(String cn_name) {
        cn_name = cn_name.trim();
        if (cn_name.length() > 4) {
            cn_name = cn_name.substring(0, 4);
            if (cn_name.endsWith("等")) {
                cn_name = cn_name.substring(0, 3);
            }
        }
        return cn_name;
    }

    public static String genLike(String colName, String bankuai) {
        List<String> list = new ArrayList<>();
        for(int i=0;(i+2)<=bankuai.length();i+=2) {
            list.add(bankuai.substring(i, i + 2));
        }
        StringBuilder result = new StringBuilder();
        for (int i = 0; i < list.size(); i++) {
            String like = list.get(i);
            result.append(" ").append(colName).append(" like '%").append(like).append("%'");
            if (i < list.size() - 1) {
                result.append(" or");
            }
        }
        return result.toString();
    }
}

File: app/src/main/java/com/wzf/wucarryme/common/utils/StringUtil.kt
package com.wzf.wucarryme.common.utils

import java.util.*

/**
 * @author wzf
 * @date 2018/3/22
 */

object StringUtil {

    fun banKuai(str: String): String {
        var str = str

        str = str.replace("类".toRegex(), "")
        str = str.replace("概念".toRegex(), "")
        str = str.replace("房地产".toRegex(), "地产")
        str = str.replace("次新".toRegex(), "新")
        val index = str.indexOf("股")
        if (index > -1) {
            str = str.substring(0, index)
        }
        return str
    }

    fun getBlogTime(p: String): String? {
        val blogTime: String?
        p.replace("[\\u00A0]+", " ")
//        val endIndex = if (p.contains(" ")) p.indexOf(" ") else p.indexOf(160)
        val endIndex = p.indexOf(" ")
        blogTime = try {
            p.substring(0, endIndex)
        } catch (e: Exception) {
            null
        }
        return blogTime
    }

    fun commaStr2List(commaStr: String): List<String> {
        return Arrays.asList(*commaStr.split(",".toRegex()).dropLastWhile({ it.isEmpty() }).toTypedArray())
    }

    fun list2commaStr(list: List<String>): String {
        var result = StringBuilder()
        for (i in list.indices) {
            result.append(list[i]).append(",")
        }
        if (result.toString().endsWith(",")) {
            result = StringBuilder(result.substring(0, result.length - 1))
        }
        return result.toString()
    }

    fun formatCNName(cn_name: String): String {
        var cn_name = cn_name
        cn_name = cn_name.trim { it <= ' ' }
        if (cn_name.length > 4) {
            cn_name = cn_name.substring(0, 4)
            if (cn_name.endsWith("等")) {
                cn_name = cn_name.substring(0, 3)
            }
        }
        return cn_name
    }

    fun genLike(colName: String, bankuai: String): String {
        val list = ArrayList<String>()
        list.add(bankuai)
        run {
            var i = 0
            while (i + 2 <= bankuai.length) {
                list.add(bankuai.substring(i, i + 2))
                i += 2
            }
        }
        val result = StringBuilder()
        for (i in list.indices) {
            val like = list[i]
            result.append(" ").append(colName).append(" like '%").append(like).append("%'")
            if (i < list.size - 1) {
                result.append(" or")
            }
        }
        return result.toString()
    }
}
